microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/internal/faulttolerance/java/MicroProfileFaultToleranceASTValidator.java

```diff
@@ -274,21 +275,19 @@ private void validateRetryAnnotation(NormalAnnotation annotation)
 			Expression jitterUnitExpr = getAnnotationMemberValueExpression(
 					annotation, JITTER_DELAY_UNIT_RETRY_ANNOTATION_MEMBER);
 
-			Object delayConstant = delayExpr != null
-					? delayExpr.resolveConstantExpressionValue()
-					: null;
-			Object maxDurationConstant = maxDurationExpr != null
-					? maxDurationExpr.resolveConstantExpressionValue()
-					: null;
-			Object jitterConstant = jitterExpr != null
-					? jitterExpr.resolveConstantExpressionValue()
+			Object delayConstant = delayExpr != null ? delayExpr.resolveConstantExpressionValue() : null;
+			Object maxDurationConstant = maxDurationExpr != null ? maxDurationExpr.resolveConstantExpressionValue()
 					: null;
+			Object jitterConstant = jitterExpr != null ? jitterExpr.resolveConstantExpressionValue() : null;
 
-			int delayNum = delayConstant != null ? (int) delayConstant : -1;
-			int maxDurationNum = maxDurationConstant != null
-					? (int) maxDurationConstant
-					: -1;
-			int jitterNum = jitterConstant != null ? (int) jitterConstant : 0;
+			long delayNum = delayConstant instanceof Integer ? (long) (int) delayConstant

```

Comment:
This all works. Just one thing I noticed to maybe save a bit of space :

```
jshell> Integer int_val = 3;
int_val ==> 3
jshell> long res = (long) (int) int_val
res ==> 3
jshell> long res = (long) int_val
res ==> 3

jshell> Long long_val = 3L;
long_val ==> 3
jshell> long res = long_val;
res ==> 3
```

I guess the interesting part is that auto-boxing need not happen between the same types.

Reply:
Oops, so if it does work, you can probably simplify this a lot, and just assign the value if `Integer` or `Long`.

Reply:
The fact that we're coming form `Object` makes a difference. I guess we shouldn't try to play too much with the auto-boxing feature.

```
jshell> Object delay1 = 3;
delay1 ==> 3

jshell> Integer delay2 = 4;
delay2 ==> 4

jshell> (long) delay1
|  Exception java.lang.ClassCastException: class java.lang.Integer cannot be cast to class java.lang.Long (java.lang.Integer and java.lang.Long are in module java.base of loader 'bootstrap')
|        at (#12:1)

jshell> (int) delay1
$13 ==> 3

jshell> (long) delay2
$14 ==> 4

jshell> (int) delay2
$15 ==> 4
```